internal/file/file_plugin.go

```diff
@@ -198,14 +198,31 @@ func (fp *FilePlugin) handleConfigApplyRequest(ctx context.Context, msg *bus.Mes
 		)
 
 		fp.messagePipe.Process(ctx, &bus.Message{Topic: bus.DataPlaneResponseTopic, Data: response})
-		fp.handleConfigApplyFailedRequest(ctx, &bus.Message{
-			Topic: bus.ConfigApplyFailedTopic,
-			Data: &model.ConfigApplyMessage{
-				CorrelationID: correlationID,
-				InstanceID:    configApplyRequest.GetOverview().GetConfigVersion().GetInstanceId(),
-				Error:         err,
-			},
-		})
+
+		err := fp.fileManagerService.Rollback(ctx, configApplyRequest.GetOverview().GetConfigVersion().GetInstanceId())
+		if err != nil {
+			// what to do?
+			response = fp.createDataPlaneResponse(
+				correlationID,
+				mpi.CommandResponse_COMMAND_STATUS_FAILURE,
+				"Config apply failed, rollback failed",
+				configApplyRequest.GetOverview().GetConfigVersion().GetInstanceId(),
+				err.Error(),
+			)
+			fp.messagePipe.Process(ctx, &bus.Message{Topic: bus.DataPlaneResponseTopic, Data: response})
+			return
+		}
+
+		response = fp.createDataPlaneResponse(
```

Comment:
Sending a `...STATUS_FAILURE` response here with no `error` field seems wrong, but it is to indicate that the rollback was successful

Reply:
the original error that caused the config apply to fail could be sent here possibly, but not sure what others think